python_modules/dagster/dagster/_core/executor/multiprocess.py

```diff
@@ -213,6 +213,18 @@ def execute(self, plan_context, execution_plan):
                             else:
                                 yield event_or_none
                                 active_execution.handle_event(event_or_none)
+                                if event_or_none.is_resource_init_failure:
+                                    step_failure_event = DagsterEvent.step_failure_event(
+                                        step_context=plan_context.for_step(
+                                            active_execution.get_step_by_key(key)
+                                        ),
+                                        step_failure_data=StepFailureData(
+                                            error=event_or_none.event_specific_data.error,
+                                            user_failure_data=None,
+                                        ),
+                                    )
+                                    yield step_failure_event
+                                    active_execution.handle_event(step_failure_event)
```

Comment:
so the choice here is to emit step failure without step start for init failure? Did you consider other approaches? 

If we pursue this approach, should we respect the retry policy?

Reply:
If we took this approach, then I would think no. The lack of step start is a good point, and I wonder if we could better solve this by just moving step start to occur before initialization occurs.